src/block/WaterCauldron.php

```diff
@@ -128,9 +129,10 @@ public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player
 			}else{
 				$this->mix($item, VanillaItems::GLASS_BOTTLE(), $returnedItems);
 			}
-		}elseif($item instanceof Armor){
+		}elseif($item instanceof Armor || $item instanceof HorseArmor){
 			if($this->customWaterColor !== null){
 				if(match($item->getTypeId()){ //TODO: a DyeableArmor class would probably be a better idea, since not all types of armor are dyeable
```

Comment:
Could we just use Tags instead a new class for this TODO?

but that's nothing for this pr anyway